app/models/solidus_stripe/gateway.rb

```diff
@@ -85,17 +85,15 @@ def capture(amount_in_cents, payment_intent_id, options = {})
     end
 
     # Authorizes and captures a certain amount on the provided payment source.
-    def purchase(amount_in_cents, _source, options = {})
-      currency = options.fetch(:currency)
-
+    # @todo add support for purchasing custom amounts
+    def purchase(_amount_in_cents, source, options = {})
       # Charge the Customer instead of the card:
-      payment_intent = request do
-        Stripe::PaymentIntent.create({
-          amount: to_stripe_amount(amount_in_cents, currency),
-          currency: currency,
-          **options[:payment_intent_options].to_h
-        })
-      end
+      payment_intent =
+        SolidusStripe::PaymentIntent.create_stripe_intent(
```

Comment:
> I'm not sure about the gain of splitting the find_or_create into find || create

@elia how can we call create_stripe_intent here without splitting? I mean, we can but already knowing the find part won't retrieve anything and it will go in the create path of the method. Is that what you are proposing?

Reply:
We could still have the `retrieve_or_create...` method and use it only when rendering the Stripe element, while using the new `create...` from the gateway (otherwise the code reader might be confused wondering in which situation a payment intent could already be present). However, I'm not sure that we should make it for a single-use case.

Reply:
Happy to change it, though, if you think that's a good idea.